astra/src/main/java/com/slack/astra/bulkIngestApi/BulkIngestS3Producer.java

```diff
@@ -0,0 +1,197 @@
+package com.slack.astra.bulkIngestApi;
+
+import com.slack.astra.metadata.dataset.DatasetMetadataStore;
+import com.slack.astra.proto.config.AstraConfigs;
+import com.slack.astra.proto.wal.WalProtos;
+import com.slack.service.murron.trace.Trace;
+import io.micrometer.core.instrument.Counter;
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.Timer;
+import java.time.Instant;
+import java.time.ZoneOffset;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.apache.kafka.clients.producer.RecordMetadata;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import software.amazon.awssdk.core.async.AsyncRequestBody;
+import software.amazon.awssdk.services.s3.S3AsyncClient;
+import software.amazon.awssdk.services.s3.model.DeleteObjectRequest;
+import software.amazon.awssdk.services.s3.model.PutObjectRequest;
+
+public class BulkIngestS3Producer extends BulkIngestProducer {
+
+  private static final Logger LOG = LoggerFactory.getLogger(BulkIngestS3Producer.class);
+  protected final String kafkaTopic;
+
+  protected final S3AsyncClient s3Client;
+  protected final String walBucket;
+  private final Counter s3UploadCounter;
+  private final Counter s3SpansUploadedCounter;
+  private final Timer s3UploadTimer;
+  private final Counter s3BytesUploadedCounter;
+
+  public BulkIngestS3Producer(
+      final DatasetMetadataStore datasetMetadataStore,
+      final AstraConfigs.PreprocessorConfig preprocessorConfig,
+      final MeterRegistry meterRegistry,
+      S3AsyncClient s3Client) {
+
+    super(datasetMetadataStore, preprocessorConfig, meterRegistry, s3Client);
+
+    // Initialize S3Producer specific fields
+    this.s3Client = s3Client;
+    this.walBucket = preprocessorConfig.getS3WalConfig().getS3Bucket();
+    this.kafkaTopic = preprocessorConfig.getKafkaConfig().getKafkaTopic();
+    this.s3UploadCounter = meterRegistry.counter("s3_wal_uploads_total");
+    this.s3SpansUploadedCounter = meterRegistry.counter("s3_wal_spans_uploaded_total");
+    this.s3UploadTimer = meterRegistry.timer("s3_wal_upload_duration");
+    this.s3BytesUploadedCounter = meterRegistry.counter("s3_wal_bytes_uploaded_total");
```

Comment:
create constants for the metric names here ala the BulkIngestProducer. Also, they should probably be prefixed with `bulk_ingest_producer` so it's clear what they are a part of.